plugins/audio-s5gen2/fu-audio-s5gen2-device.c

```diff
@@ -47,88 +48,96 @@ fu_qc_s5gen2_device_msg_out(FuQcS5gen2Device *self, guint8 *data, gsize data_len
 }
 
 static gboolean
-fu_qc_s5gen2_device_msg_in(FuQcS5gen2Device *self, guint8 *data_in, gsize data_len, GError **error)
+fu_qc_s5gen2_device_msg_in(FuQcS5gen2Device *self,
+			   guint8 *data_in,
+			   gsize data_len,
+			   gsize *read_len,
+			   GError **error)
 {
 	FuDevice *proxy = fu_device_get_proxy(FU_DEVICE(self));
+	g_autoptr(GByteArray) err_msg = NULL;
+	g_autoptr(GError) error_local = NULL;
+
 	if (proxy == NULL) {
 		g_set_error_literal(error, FWUPD_ERROR, FWUPD_ERROR_NOT_SUPPORTED, "no proxy");
 		return FALSE;
 	}
-	return fu_qc_s5gen2_impl_msg_in(FU_QC_S5GEN2_IMPL(proxy), data_in, data_len, error);
+	if (!fu_qc_s5gen2_impl_msg_in(FU_QC_S5GEN2_IMPL(proxy), data_in, data_len, read_len, error))
+		return FALSE;
+
+	err_msg = fu_struct_qc_error_ind_parse(data_in, *read_len, 0, &error_local);
```

Comment:
i'm a bit confused here -- the 2nd parameter is supposed to be the length of the 1st thing.

Reply:
It should be correct due `fu_qc_s5gen2_impl_msg_in()` above.

I'd prefer to leave as is, just in case if the device return data lesser than structure size, so it would help to avoid potential (about to impossible I think) false positive here.

Reply:
What if `read_len` is returned bigger than `data_in_len`? We'd get an out of memory read.

Reply:
@hughsie and you're right. Immediately catch one issue with it.
thanks!